app/services/admin/data_counts.rb

```diff
@@ -0,0 +1,30 @@
+module Admin
+  class DataCounts
+    def self.call
+      # If this class is to be re-used in more situations, we could pass in params
+      # to dictate which counts are needed, to reduce unneeded queries.
+
+      open_abuse_reports_count =
+        FeedbackMessage.open_abuse_reports.size
+
+      possible_spam_users_count = User.registered.where("length(name) > ?", 30)
+        .where("created_at > ?", 48.hours.ago)
+        .order(created_at: :desc)
+        .select(:username, :name, :id)
+        .where.not("username LIKE ?", "%spam_%")
+        .size
+
+      flags = Reaction
+        .includes(:user, :reactable)
+        .privileged_category
+      {
```

Comment:
Blocking, this looks like a great case for returning a Struct object, instead of relying on hash access.